File: 3rd_Semester/DSA/Lab4/towerOfHanoi.cpp
#include <iostream>
using namespace std;

void towerOfHanoi(int n, string source, string temp, string destination)
{
    if (n == 1)
    {
        cout << "Move disk 1 from " << source << " to " << destination << endl;
        return;
    }
    
    towerOfHanoi(n - 1, source, destination, temp);
    cout << "Move disk " << n << " from " << source << " to " << destination << endl;
    towerOfHanoi(n - 1, temp, source, destination);
}

// Driver code
int main()
{
    int n; // Number of disks
    cout << "Enter number of disks: ";
    cin >> n;
    towerOfHanoi(n, "Source", "Temp", "Destination"); // A, B and C are names of rods
    return 0;
}

File: 4th_Semester/AI/tower_of_hanoi.py
# Python function to solve the tower of hanoi

def toh(n, source, destination, temp):
    if n == 1:
        print(f"Move disk 1 from {source} to {destination}")
        return
    toh(n - 1, source, temp, destination)
    print(f"Move disk {n} from {source} to {destination}")
    toh(n-1, temp, destination, source)


# Enter the number of rods
n = int(input("Enter the number of rods: "))
toh(n, 'Source', 'Destination', 'Temp')
